client/src/components/Workflow/Run/WorkflowRunName.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BAlert } from "bootstrap-vue";
+import { onMounted, ref } from "vue";
+
+import { getWorkflowInfo } from "@/components/Workflow/workflows.services";
+
+import { errorMessageAsString } from "../../../utils/simple-error";
+
+import WorkflowInformation from "@/components/Workflow/Published/WorkflowInformation.vue";
+
+const props = defineProps({
+    model: {
+        type: Object,
+        required: true,
+    },
+});
+
+const loading = ref(true);
+const messageText = ref<string | null>(null);
+const messageVariant = ref<string | null>(null);
+const expandAnnotations = ref(true);
+const workflowInfoData = ref(null);
+
+onMounted(() => {
+    loadAnnotation();
+});
+
+async function loadAnnotation() {
+    loading.value = true;
+
+    try {
+        const workflowInfoDataPromise = getWorkflowInfo(props.model.runData.id);
```

Comment:
This bypasses all stores and fetches the workflow even if we already have it. Can you use useWorkflowInstance instead ?

An even better alternative is to serialize the workflow annotation for the run request. This would be as simple as
```diff
diff --git a/lib/galaxy/managers/workflows.py b/lib/galaxy/managers/workflows.py
index 0d970de242..5921ad74f4 100644
--- a/lib/galaxy/managers/workflows.py
+++ b/lib/galaxy/managers/workflows.py
@@ -1051,6 +1051,7 @@ class WorkflowContentsManager(UsesAnnotations):
             step_models.append(step_model)
         return {
             "id": trans.app.security.encode_id(stored.id),
+            "annotation": stored.annotation,
             "history_id": trans.app.security.encode_id(history.id) if history else None,
             "name": stored.name,
             "owner": stored.user.username,
```

we only use the run endpoint here in the workflow run component, so if we always need the attribute I don't see a reason not to just serialize that as well.

You would then have this available in the runData object here.

Reply:
_Follow-up questions related to the [existing _workflow_to_dict_run() method](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/managers/workflows.py#L984) in Workflows.py (eg. maybe some refactoring needed as a separate ticket)._

**(1) Blocker: I am assuming the purpose of @mvdbeek 's code suggestions is to _not_ make an additional database call, correct? However, using the VSCode Debugger to find already-available values, I have not yet been able to find the following variables for the WorkflowInformation.vue component to be fully displayed. Specifically, `creator`, `email_hash`. and `license`. Although, the License may not be appearing because there is no value for it in my dev-qa use-case, but `email_hash` should display (since there is an avatar selected) and `creator` should always have a value. I am assuming that Email Hash can be created from the non-hashed email string parameter: stored.annotations[0].stored_workflow.user.email?** I was able to find/serialize the other values (including Workflow Tags).
(2) Side note: calling `stored.annotation` (singular) fails with the error: 'StoredWorkflow' object has no attribute 'annotation'. The plural value can be accessed using `stored.annotations` ; however, it lacks parameters like the Workflow Tags. Instead, I used `stored.annotations[0].stored_workflow` to get the Workflow Tags - if that is an unreliable value, please let me know.
(3) Side note: using `useWorkflowInstance` pulls up an entirely different Workflow when using the Encoded Workflow ID (that appears in the URL). Tried also with the literal Workflow ID number (as a string) and failed. Perhaps this works only for Invocations since that is where I see `useWorkflowInstance` used in the codebase currently...? The method seemed to be able to used successfully in `WorkflowInvocationHeader.vue` so hopefully @ahmedhamidawan can please advise on this question?
(4) Question: Not a blocker, but I was surprised the following element didn't contain Workflow Annotation data within the same method: `workflow.annotation` returns `null` instead of any actual Workflow Annotation data. If that is not expected, then this line is broken. If "null" is expected, how can this be -is the variable misnamed? [This code block](https://github.com/galaxyproject/galaxy/blame/dev/lib/galaxy/managers/workflows.py#L1015-L1018) is pretty old, but perhaps last editors have some thoughts: @guerler @nsoranzo @dannon ?

Reply:
`useWorkflowInstance` takes the stored `workflow_id` as parameter, not the `encoded_workflow_id` we have here from the run data. The stored id is essentially the id for the specific workflow (for the given version; latest in this case) retrieved here:
https://github.com/galaxyproject/galaxy/blob/aa2574343555f10294dc626f991d7b728dfccb5a/lib/galaxy/managers/workflows.py#L916

Once we have that id, the whole `loadAnnotation` method could be replaced by a:
```
const { workflow, loading, error } = useWorkflowInstance(stored_workflow_id);
```

Reply:
@ahmedhamidawan Thank you so much for this helpful comment. The latest iteration of this (v3) doesn't need this additional data right now, but I may need to know this helpful piece of information in the future.